### ISSUE TITLE
Dictionary Support

### ISSUE THREAD

**Issue by paddor** [foxglove/wasm-zstd#2]
As the title says.

**Comment by defunctzombie** [comment]
@paddor Is there a bug you would like to describe or something that is not working? We reserve the bug label for functionality that we expect to work but is not working. If you could describe what you tried to do and what did not work that would be helpful to furthering whether this issue is a bug.

**Comment by paddor** [comment]
@defunctzombie No, it's not a bug. It's a feature request but I opened it here because I'm not familiar with the Discussions page that it linked me to. It's very simple: Is Zstd dictionary support planned?

**Comment by jhurliman** [comment]
@paddor this library was primarily created to facilitate simple compression and decompression of zstd data in the MCAP (https://mcap.dev/) and Foxglove Studio (https://github.com/foxglove/studio) TypeScript codebases, so there is currently no dictionary support. Having said that, we're open to pull requests if someone wants to add additional zstd functionality. It boils down to adding a new C method (example: https://github.com/foxglove/wasm-zstd/blob/main/src/wasm-zstd.c#L33-L40) and wrapping JS method (example: https://github.com/foxglove/wasm-zstd/blob/main/src/index.js#L57-L870) where most of the complexity is on the JS side for malloc/free and marshaling data in and out of the WASM boundary.
